src/lib.rs

```diff
@@ -124,11 +124,13 @@ pub enum Error {
     Fragmented,
     /// An incoming packet was recognized but was self-contradictory.
     /// E.g. a TCP packet with both SYN and FIN flags set.
+    /// Or a IGMP packet querying address that is not multicast
```

Comment:
The above is just an example, you don't need to add every conceivable error condition.